File: src/main/java/me/nylestroke/nylemod/entity/client/block/GoldenStandModel.java
package me.nylestroke.nylemod.entity.client.block;

import me.nylestroke.nylemod.NylemodExample;
import net.minecraft.util.Identifier;
import software.bernie.geckolib3.model.AnimatedGeoModel;

public class GoldenStandModel extends AnimatedGeoModel<GoldenStandEntity> {

    @Override
    public Identifier getModelLocation(GoldenStandEntity object) {
        return new Identifier(NylemodExample.MOD_ID, "geo/golden_stand.geo.json");
    }

    @Override
    public Identifier getTextureLocation(GoldenStandEntity object) {
        return new Identifier(NylemodExample.MOD_ID, "textures/machines/golden_stand.png");
    }

    @Override
    public Identifier getAnimationFileLocation(GoldenStandEntity animatable) {
        return new Identifier(NylemodExample.MOD_ID, "animations/golden_stand.animation.json");
    }

}

File: src/main/kotlin/me/nylestroke/nylemod/block/entity/client/GoldenStandModel.kt
package me.nylestroke.nylemod.block.entity.client

import me.nylestroke.nylemod.NylemodExample
import me.nylestroke.nylemod.block.entity.GoldenStandEntity
import net.minecraft.util.Identifier
import software.bernie.geckolib3.model.AnimatedGeoModel

class GoldenStandModel : AnimatedGeoModel<GoldenStandEntity?>() {
    override fun getModelResource(`object`: GoldenStandEntity?): Identifier {
        return Identifier(NylemodExample.MOD_ID, "geo/golden_stand.geo.json")
    }

    override fun getTextureResource(`object`: GoldenStandEntity?): Identifier {
        return Identifier(NylemodExample.MOD_ID, "textures/machines/golden_stand.png")
    }

    override fun getAnimationResource(animatable: GoldenStandEntity?): Identifier {
        return Identifier(NylemodExample.MOD_ID, "animations/golden_stand.animation.json")
    }
}